reliable-webservice-go/README.md

```diff
@@ -0,0 +1,71 @@
+# Day 3: Building Reliable Web Services
+
+These are the code-samples, exercises, quizzes, and links needed for this course. To complete these exercises follow the instructions in this doc. In the subdirectories of this section are example solutions. Your solutions do not need to match the provided solutions. The goal of the exercises is to learn and understand the why things are implemented so that a participant apply principles to any framework or problem set.
+
+## Exercises
+
+### [Exercise 1](/ex-1-auth/auth.go) Add auth to your server endpoints
+
+Auth tooling is sometimes the first or last measure of security for your endpoints. There are various methods for adding auth to your server endpoint. When building production services the methods you choose for authentiction and authorization will be determined by security professionals, but how you implement them is up to you as a developer.
+
+In this exercise add an auth method to your server. You can use middle-ware to add auth to all endpoints, but functions to add it to specific endpoints or, add it manualy to just one. Here are some examples of how to add different kinds of auth.
+
+* [chi middleware](https://github.com/go-chi/chi/blob/master/middleware/basic_auth.go)
+* [manual basic token](https://github.com/Soypete/golang-cli-game/blob/main/server/helpers.go#L36)
+* [other-JWT](https://pkg.go.dev/github.com/golang-jwt/jwt/v5#example-Parse-Hmac)
+
+### [Exercise 2](/ex-2-middleware/middleware.go) Add middleware to your go server
+
+Middleware allows you add complex funcitonality to your endpoints. You can use middle ware to add metrics, auth, proffiling or custom logic to your programs.
+
+Using one of the web frameworks or the go standard library add a middle ware function to your server.
+Web frameworks:
+
```

Comment:
I kind of feel like these two exercises should be swapped. Exercise 2 introduces them to middleware and practices with a basic use case. Then exercise 1 builds on it showing how you can implement auth using middleware.

Reply:
Yes, but in the course middle ware is after auth, and the rubric is published. This is the problem with building the course after planning it